packages/api/src/services/profileValidation.js

```diff
@@ -20,7 +21,10 @@ function profileValidation(user) {
   }
   if (user.role.toLowerCase() === 'student') {
     if (!Reflect.has(user, 'batchId') || user.batchId.length < 24) {
```

Comment:
What if the batchId is > 24? That also is an invalid MongoDB ObjectId.

When you have time, look into the tools mongoose.js gives you to deal with validation errors (see the tests in /packages/api/src/db/__tests__/user.js for some examples; you'll have to search their docs for how they deal with foreign key validation)

A lot of code in this file is stuff that mongoose is already running for us.